tmt/utils.py

```diff
@@ -1585,16 +1586,16 @@ def dict_to_yaml(
         data: Union[Dict[str, Any], List[Any], 'tmt.base._RawFmfId'],
         width: Optional[int] = None,
         sort: bool = False,
-        start: bool = False) -> str:
+        start: Optional[bool] = False) -> str:
```

Comment:
Now, this is suspicious. Why is `Optional` suddenly needed? I bet it's related to what's inside `yaml` object below, but, I'd prefer `start` to remain `bool` - our code would never try to use `start=None`, it'd always be a `bool`, therefore if we need to mend it with YAML parser internals, let's do it inside this function.

I suppose `ignore[assignment]` below is related, what does mypy reports for these lines?

Reply:
My bad, thanks for catching it. I was trying to match the ruamel initially, but yes, with the ignore there it passes.